### DISCUSSION TITLE
Disable spiking behaviour of output layer spiking neurons in multi-step-mode

### DISCUSSION THREAD

**Discussion by gwgkarunasena57** [fangwei123456/SpikingFlow#553] [Q&A]
Hi,

How can i disable the the spikes of output layer LIF neurons of the SNN, when the network is in multi-step-mode?
Thank you.

Thanks and Rgds,
Udayanga

**Comment by shayanjl** [comment]
In SpikingJelly, a `LIFNode` is a spiking neuron, so in multi-step mode it still outputs spikes over time. If you want the output layer to stop firing and instead return a continuous signal, the usual approach is to **replace the last LIF layer with a non-spiking output layer**, for example a synapse/filter layer or a custom “NonSpikingLIFNode” that updates the membrane voltage but skips `neuronal_fire()` and `neuronal_reset()`. The docs describe this idea as equivalent to a LIF neuron with an infinite threshold, and specifically note that it can be used in the last layer. Multi-step behavior is controlled by `step_mode='m'`. ([[SpikingJelly](https://spikingjelly.readthedocs.io/zh-cn/latest/_modules/spikingjelly/activation_based/layer/misc.html?utm_source=chatgpt.com)][1])
